project/Boilerplate.scala

```diff
@@ -0,0 +1,410 @@
+        -    val tree = 
+        -    q\"\"\"$$self.combineIntTileCombiner(new geotrellis.macros.IntTileCombiner${arity} {
+        -       ${quoted}
+        -       def apply(${argsInt}): Int = $$f(${seq})
+        -    })\"\"\"
+        -    new InlineUtil[c.type](c).inlineAndReset[T](tree)
+        -  }
+        -  def doubleCombine${arity}_impl[T, MBT <: MacroCombinableMultiBandTile[T]](c: Context)(${exprSeqInt})(f: ${exprFArgsDouble}): c.Expr[T] = {
+        -    import c.universe._
+        -    val self = c.Expr[MacroCombinableMultiBandTile[T]](c.prefix.tree)
+        -    val tree = 
+        -    q\"\"\"$$self.combineDoubleTileCombiner(new geotrellis.macros.DoubleTileCombiner${arity} {
+        -       ${quoted}
+        -       def apply(${argsDouble}): Double = $$f(${seq})
+        -    })\"\"\"
+        -    new InlineUtil[c.type](c).inlineAndReset[T](tree)
+        -  }       
+         |}        
+      """
+    }
+  }  
+
+  object GenMacroCombineFunctions extends Template {    
+    def filename(root: File) = root / "geotrellis" / "macros" / "MacroCombineFunctions.scala"
+    override def range = 3 to maxArity
+    def content(tv: TemplateVals) = {
+      import tv._
+
+      val argsInt    = typedSeq("Int")
+      val argsDouble = typedSeq("Double")
+
+      def seqFunc(ts: String) = s"(${namedSeq(ts)}) => $ts"
+      def seqFInt             = seqFunc("Int")
+      def seqFDouble          = seqFunc("Double")
+
+      block"""
+         |package geotrellis.macros
+         |import scala.language.experimental.macros
+         |trait MacroCombineFunctions[T, MBT <: MacroCombinableMultiBandTile[T]] {
+        -  def combine(${argsInt})(f: ${seqFInt}): T =
+        -    macro MultiBandTileMacros.intCombine${arity}_impl[T, MBT]
+        -  def combineDouble(${argsInt})(f: ${seqFDouble}): T =
+        -    macro MultiBandTileMacros.doubleCombine${arity}_impl[T, MBT]
+         |}        
+      """
+    }
+  }
+
+  object GenMacroMultibandCombiners extends Template {
+    def filename(root: File) = root / "geotrellis" / "raster" / "MacroMultibandCombiners.scala"
+    override def range = 3 to maxArity
+    def content(tv: TemplateVals) = {
+      import tv._
+
+      val bandVals = (0 until arity) map { i => s"val band$i = band(combiner.b$i)" } mkString "; "
+      val bandIntArgs = (0 until arity) map { i => s"band$i.get(col, row)" } mkString ", "
+      val bandDoubleArgs = (0 until arity) map { i => s"band$i.getDouble(col, row)" } mkString ", "
+
+      block"""
+         |package geotrellis.raster
+         |import geotrellis.macros._
+         |import spire.syntax.cfor._              
+         |trait MacroMultibandCombiners { self: MultiBandTile =>      
+        -  def combineIntTileCombiner(combiner: IntTileCombiner${arity}): Tile = {
+        -    ${bandVals}
+        -    val result = ArrayTile.empty(cellType, cols, rows)
+        -    val arr = Array.ofDim[Int](bandCount)
```

Comment:
I am curious about this line, think we don't need it, `arr` has no usages in the function body.